libraries/cbor/src/values.rs

```diff
@@ -124,16 +123,35 @@ impl Ord for Value {
             (ByteString(b1), ByteString(b2)) => b1.len().cmp(&b2.len()).then(b1.cmp(b2)),
             (TextString(t1), TextString(t2)) => t1.len().cmp(&t2.len()).then(t1.cmp(t2)),
             (Array(a1), Array(a2)) if a1.len() != a2.len() => a1.len().cmp(&a2.len()),
+            (Array(a1), Array(a2)) => {
+                // Arrays of same length.
+                let mut ordering = Ordering::Equal;
+                for (e1, e2) in a1.iter().zip(a2.iter()) {
+                    ordering = ordering.then(e1.cmp(e2));
+                    if !matches!(ordering, Ordering::Equal) {
+                        break;
+                    }
+                }
+                ordering
+            }
             (Map(m1), Map(m2)) if m1.len() != m2.len() => m1.len().cmp(&m2.len()),
+            (Map(m1), Map(m2)) => {
+                // Maps of same length.
+                let mut ordering = Ordering::Equal;
+                for ((k1, v1), (k2, v2)) in m1.iter().zip(m2.iter()) {
+                    ordering = ordering.then(k1.cmp(k2));
+                    ordering = ordering.then_with(|| v1.cmp(v2));
```

Comment:
And this is the same as `k1.cmp(k2).then_with(|| v1.cmp(v2))`

Reply:
Done